### DISCUSSION TITLE
How to simply pass a subdocument reference to a function ?

### DISCUSSION THREAD

**Discussion by bmedici** [nlohmann/json#4107] [Q&A] [answered]
Maybe my question will seem basic, but I'm trying to do this I cant find the right type and syntax do succeed:

```
void handler(TYPE& doc) {
   doc["value"] = 6;
}

json ex1 = json::parse(R"(
  {
    "pi": 3.141,
    "happy": true
  }
)");

subdoc = ex1["sd1"];
handler(subdoc);
```

Thank you !

**Comment by PatchByte** [comment]
Try:
```c++
void handler(json& doc) {
   doc["value"] = 6;
}

json ex1 = json::parse(R"(
  {
    "pi": 3.141,
    "happy": true
  }
)");

subdoc = ex1["sd1"];
handler(subdoc);
```

**Reply by PatchByte** [reply]
But be careful and check if the "sd1" field exists!

**Reply by bmedici** [reply]
Thank you, by just "mentionning" it, I thought it would have been created. Then I should initialise it to an empty object, or array, or value, right ?

**Reply by PatchByte** [reply]
yes you can just initialize another sub json. 
In this case you can do:
```c++
void handler(json& doc) {
   doc["value"] = 6;
}

json ex1 = json::parse(R"(
  {
    "pi": 3.141,
    "happy": true
  }
)");

if(ex1["sd1"].is_null() == true)
{
 // Way 1:
 ex1["sd1"] = {};
 // Way 2:
 ex1["sd1"] = json();
 // Whatever you like
}

subdoc = ex1["sd1"];
handler(subdoc);
```

**Reply by PatchByte** [reply]
Also another thing, if you don't want to use json::parse for json structures in the code you can just do:
```c++
json ex1 = {
 { "pi": 3.141f },
 { "happy": true }
};
```

**Comment by gregmarr** [comment, marked as answer]
The type is the same as the original object, so in this case, `json`.

**Reply by bmedici** [reply]
That was so obvious that I did not think about that, thank you, going to try this. What a great API !
